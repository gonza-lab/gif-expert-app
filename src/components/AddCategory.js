import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setIValue] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prevCat) => [inputValue, ...prevCat]);
      setIValue('');
    }
  };

  const handleChange = (e) => {
    setIValue(e.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
