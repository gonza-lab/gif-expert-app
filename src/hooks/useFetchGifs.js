import { useState, useEffect } from 'react';
import { getGif } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [data, setData] = useState({
    loading: true,
    images: [],
  });

  useEffect(() => {
    getGif(category).then((gifs) =>
      setData({
        loading: false,
        images: gifs,
      })
    );
  }, []);

  return data;
};
