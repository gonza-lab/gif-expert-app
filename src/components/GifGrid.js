import React /* , { useEffect, useState } */ from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { loading, images } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p className="animate__animated animate__flash">Cargando</p>}
      <div className="gifGrid">
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
