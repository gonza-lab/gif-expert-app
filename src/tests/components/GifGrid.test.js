import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import React from 'react';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en GifGid', () => {
  test('Debe mostrar el componente correctamente', () => {
    useFetchGifs.mockReturnValue({
      loading: true,
      images: [],
    });

    const wrapper = shallow(<GifGrid category="Jest" />);

    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar imagenes cuando se cargan los gifs', () => {
    const images = [
      {
        id: 'ABC',
        url: 'https://slash/caulquiercosa.jpg',
        title: 'cualquiercosa',
      },
      {
        id: 'ABCD',
        url: 'https://slash/caulquiercosa.jpg',
        title: 'cualquiercosa',
      },
    ];
    useFetchGifs.mockReturnValue({
      loading: false,
      images,
    });

    const wrapper = shallow(<GifGrid category="Jest" />);
    expect(wrapper.find('p').exists()).not.toBeTruthy();
    expect(wrapper.find('GifGridItem').length).toBe(images.length);
  });
});
