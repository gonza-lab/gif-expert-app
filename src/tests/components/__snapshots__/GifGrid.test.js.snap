// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGid Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Jest
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="gifGrid"
  />
</Fragment>
`;
